nrf_security/doc/different_tfm_config.rst

```diff
@@ -0,0 +1,62 @@
+.. _nrf_security_different_tfm_config:
+
+Different configurations of secure and non-secure
+#################################################
+
+Normally in a TF-M application the non-secure app will invoke the PSA
+Crypto API to do crypto. And the feature set of the PSA Crypto API
+will be configured through nrf_security. So there will be no crypto
+implementations on the non-secure side.
+
+Therefore in a normal TF-M application there will be two instances of
+nrf_security, one in the non-secure image that just contains an API
+for PSA Crypto. And one in the secure image which contains a
+cryptocell driver and possibly other backends as well. Yet they will
+both use the same configuration.
+
+But, if an application needs to have some crypto features implemented
+in the non-secure nrf_security instance and other features implemented
+in the secure nrf_security instance the following guide can be
+followed to configure each instance independently:
+
+1. Build the application for a non-secure board with TF-M.
+
+   .. code-block:: console
+
+      export APP_DIR=~/ncs/nrf/samples/crypto/psa_tls
+      cd $APP_DIR
+      west build -b nrf9160dk_nrf9160_ns
+
+#. Enable nrf_security, if it is not enabled already, and configure
+   the crypto functionality and backends as they should be configured
+   on the secure side. See :ref::`_nrf_security_backend_config` for
+   configuration of nrf_security.
+
+#. Copy the resulting configuration from the build directory to the
+   application source directory. This copy will be used to configure
+   the secure nrf_security instance.
+
+   .. code-block:: console
+
+      cp $APP_DIR/build/zephyr/.config $APP_DIR/secure_zephyr.config
+      mkdir -p $APP_DIR/secure/zephyr
+      cp $APP_DIR/build/zephyr/include/generated/autoconf.h $APP_DIR/secure/zephyr/
```

Comment:
Could we provide a Kconfig option for this, and then use a `add_custom_command(OUTPUT <path>/secure/zephyr/autoconf.h)` without having a `DEPENDS`.

That way `add_custom_command()` will only copy the file if it is missing, but not overwrite an already existing / modified file.

Will save the reader some work, and at the same time allow local modifications.

Having such setting could potentially also remove the need for doing the `cat ... >> CMakeLists.txt` below.

Reply:
discussed offline.

Can be enhanced later.